plugins/modules/discovery_intent.py

```diff
@@ -70,9 +70,35 @@
       http_read_credential:
         description: HTTP read credentials for hosting a device
         type: dict
+        suboptions:
```

Comment:
We can rearrange the suboptions. We can list all the mandatory in the beginning...
username. -- Mandatory
password -- Mandatory
port -- Mandatory
secure -- Optional

Reply:
Addressed